plugins/modules/keycloak_clientscope_type.py

```diff
@@ -246,15 +246,22 @@ def main():
     if module._diff:
         result['diff'] = dict(before=result['existing'], after=result['proposed'])
 
-    if module.check_mode:
-        module.exit_json(**result)
-
     default_clientscopes_add = clientscopes_to_add(default_clientscopes_existing, default_clientscopes_real)
     optional_clientscopes_add = clientscopes_to_add(optional_clientscopes_existing, optional_clientscopes_real)
 
     default_clientscopes_delete = clientscopes_to_delete(default_clientscopes_existing, default_clientscopes_real)
     optional_clientscopes_delete = clientscopes_to_delete(optional_clientscopes_existing, optional_clientscopes_real)
 
+    result["changed"] = (
+        len(default_clientscopes_add) > 0
+        or len(optional_clientscopes_add) > 0
+        or len(default_clientscopes_delete) > 0
+        or len(optional_clientscopes_delete) > 0
+    )
```

Comment:
Nothing wrong with the code in the PR, however just for kicks, me personally, I would go with:
```suggestion
    result["changed"] = any(len(x) > 0 for x in [default_clientscopes_add, optional_clientscopes_add, default_clientscopes_delete, optional_clientscopes_delete])
    )
```